src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -900,9 +900,13 @@ def _unsupported_type_error(dimensions):
                     dim_name = dim.items[1].string.lower()
                     try:
                         sym = symbol_table.lookup(dim_name)
-                        if (sym.datatype.intrinsic !=
-                                ScalarType.Intrinsic.INTEGER or
-                                isinstance(sym.datatype, ArrayType)):
+                        # Explicitly forbid symbols that we *know* are not
+                        # integer scalars. (We allow symbols of
+                        # Unknown/DeferredType.)
+                        if (isinstance(sym.datatype, ArrayType) or
```

Comment:
I know you are just modifying the original logic here but this test does not seem to be robust to extension. For example, if and when we add a StructureType we would have to add this as something we know is not an integer scalar in this test. I think it would be better if we checked that it is not any of the things that are or could be integer scalars.

Reply:
Yes, that's a good point. Done.

Reply:
Much better, thanks,